src/pngwriter.h

```diff
@@ -434,6 +439,12 @@ class pngwriter
     */
    void write_png(void);
 
+
+   /* Write PNG to a buffer
+    * Generate and write PNG to memory. The PNGwriter instance will not modified and can be used after this.
+    */
+   void write_to_buffer(std::vector<unsigned char> & buffer);
```

Comment:
should be `void write_to_buffer(...) const` then

Reply:
At the moment, it is not possible, because the `filegamma_` will checked, if it is correct. Maybe we could move the check (and setting the right value) move to the setter.

https://github.com/SimeonEhrig/pngwriter/blob/9d8c6133c2d503d1b20e1089448d072ab26be26e/src/pngwriter.cc#L1002

Reply:
I moved the filegamma check to the setter so that write_to_buffer() could be changed to const.

https://github.com/SimeonEhrig/pngwriter/blob/8b1fbdebf889bf698666eaac127da1a9136fca1d/src/pngwriter.cc#L1661-L1668

As a site effect, I could also changed close() to const.